### PR TITLE
Feature/list view

### PR DESCRIPTION
**Pull request by puranban**

### CODE REVIEW

**Review by priyesh777 — APPROVED** [review]

File: src/root/components/ListView/Message/index.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    emptyMessage = 'No data available',
+    erroredMessage = 'Oops! We ran into an issue',
+    messageHidden = false,
+  } = props;
+
+  let message: React.ReactNode = messageFromProps;
+
+  if (pending) {
+    return (
+      <PendingMessage />
```
Comment (AdityaKhatri) [inline src/root/components/ListView/Message/index.tsx:37]: use this directly

File: src/root/components/ListView/Message/index.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import BlockLoading from '#components/block-loading';
+
+function PendingMessage() {
+  return (
+    <BlockLoading />
+  );
+}
+
+export interface Props {
+  className?: string;
```
Comment (AdityaKhatri) [inline src/root/components/ListView/Message/index.tsx:11]: The provided className has not been used

File: src/root/components/ListView/Message/index.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  return (
+    <div>
+      {!messageHidden && (
+        <div>
+          {message}
+        </div>
+      )}
+    </div>
+  );
+}
```
Comment (AdityaKhatri) [inline src/root/components/ListView/Message/index.tsx:58]:
```suggestion
  if(messageHidden) {
    return null;
  }
  return (
    <div ={_cs(className, styles.message)}>
          {message}
    </div>
  );
}
```

Comment (AdityaKhatri) [reply src/root/components/ListView/Message/index.tsx:58]: We'll probably need to add a certain styling to message. Talk about this with Ankit.

File: src/root/components/ListView/index.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  };
+
+function ListView<
+  D,
+  P,
+  K extends OptionKey,
+  GP extends GroupCommonProps,
+  GK extends OptionKey
+>(props: Props<D, P, K, GP, GK>) {
+
```
Comment (AdityaKhatri) [inline src/root/components/ListView/index.tsx:33]: remove empty line

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: src/root/components/ListView/Message/index.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  const {
+    className,
+    empty,
+    pending,
+    errored,
+    message: messageFromProps,
+    emptyMessage = 'No data available',
+    erroredMessage = 'Oops! We ran into an issue',
+  } = props;
+  console.log(props);
```
Comment (priyesh777) [inline src/root/components/ListView/Message/index.tsx:28]: Please remove such console.log after using it. I think "yarn typecheck" also mentions such console statements.

File: src/root/components/ListView/index.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    className,
+    errored,
+    pending,
+    data,
+    emptyMessage,
+    pendingMessage,
+    erroredMessage,
+    ...otherListProps
+  } = props;
+  const empty = !(data?.length && data.length > 0);
```
Comment (priyesh777) [inline src/root/components/ListView/index.tsx:43]: I guess we could make this a bit simpler by writing it as :-
const empty = !(data && data.length > 0 );

Comment (puranban) [reply src/root/components/ListView/index.tsx:43]: This logic is picked from deep-ui list view.

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**